File: 2423-remove-letter-to-equalize-frequency/2423-remove-letter-to-equalize-frequency.py
class Solution:
    def equalFrequency(self, word: str) -> bool:
        repeats ={}
        for c in word:
            if c not in repeats:
                repeats[c] = 0
            repeats[c] +=1
        repeatsCount =list(repeats.values());
        for i in range(len(repeatsCount)):
            currentRepate =repeatsCount[i]
            temp = list(repeatsCount)
            if (currentRepate ==1):
                temp.remove(currentRepate)
            else:
                temp[i]-=1
            if(len(set(temp)) ==1 ):
                return True
        return False
            

File: 2423-remove-letter-to-equalize-frequency/2423-remove-letter-to-equalize-frequency.cs
public class Solution {
    public bool EqualFrequency(string word) {
        Dictionary<char,int> repeats = new  Dictionary<char,int>();
        foreach(char c in word) {
            if(!repeats.ContainsKey(c))
                repeats[c] = 0;
            repeats[c]+=1;
        }
        List<int> CharsRepeats = new List<int>();
        foreach(var pair in repeats)
            CharsRepeats.Add(pair.Value);

        for(int i=0;i<CharsRepeats.Count;i++){
            List<int> temp = new List<int>(CharsRepeats);
            if(temp[i] ==1 )temp.RemoveAt(i);
            else temp[i]--;
            if (temp.Distinct().Count() == 1)
                return true;
        }
        return false;
    }
}